cubic-server/Player.cpp

```diff
@@ -619,15 +645,43 @@ void Player::_onPlayerAbilities(const std::shared_ptr<protocol::PlayerAbilities>
 void Player::_onPlayerAction(const std::shared_ptr<protocol::PlayerAction> &pck)
 {
     // LINFO("Got a Player Action ", pck->status, " at ", pck->location);
-    LDEBUG("Got a Player Action and player is in gamemode ", this->getGamemode(), " and status is ", pck->status);
-    if (pck->status == 0 && this->getGamemode() == 1) {
-        this->getDimension()->blockUpdate(pck->location, 0);
-    } else if (pck->status == 2) {
+    LDEBUG("Got a Player Action and player is in gamemode ", this->getGamemode(), " and status is ", (int32_t) pck->status);
+    switch (pck->status) {
+    case protocol::PlayerAction::Status::StartedDigging:
+        if (this->getGamemode() == 1)
+            this->getDimension()->blockUpdate(pck->location, 0);
+        break;
+    case protocol::PlayerAction::Status::CancelledDigging:
+        break;
+    case protocol::PlayerAction::Status::FinishedDigging:
         this->getDimension()->blockUpdate(pck->location, 0);
+        _foodExhaustionLevel += 0.005;
+        break;
+    case protocol::PlayerAction::Status::DropItemStack:
+        break;
+    case protocol::PlayerAction::Status::DropItem:
+        break;
+    case protocol::PlayerAction::Status::ShootArrowOrFinishEating:
+        _eat(922);
+        break;
+    case protocol::PlayerAction::Status::SwapItemInHand:
+        break;
+    default:
+        LERROR("Got a Player Action with an unknown status: ", (int32_t) pck->status);
+        break;
     }
 }
 
-void Player::_onPlayerCommand(const std::shared_ptr<protocol::PlayerCommand> &pck) { LDEBUG("Got a Player Command"); }
+void Player::_onPlayerCommand(const std::shared_ptr<protocol::PlayerCommand> &pck)
+{
+    LDEBUG("Got a Player Command");
+    if (pck->action_id == 3) {
```

Comment:
Please make this an enum when you can, this will make the code way more readable.